src/main/java/org/blockchain_innovation/factom/identiy/did/LowLevelIdentityClient.java

```diff
@@ -65,11 +66,17 @@ public EntryApi getEntryApi() {
      * @return
      * @throws RuleException
      */
-    public List<FactomIdentityEntry<?>> getAllEntriesByIdentifier(String identifier, EntryValidation validate, Optional<Long> maxHeight, Optional<Long> maxTimestamp) throws RuleException {
+    public List<FactomIdentityEntry<?>> getAllEntriesByIdentifier(String identifier, EntryValidation validate, Optional<Long> maxHeight, Optional<Long> maxTimestamp) throws RuleException, ParserException {
         List<FactomIdentityEntry<?>> entries = new ArrayList<>();
         String chainId = identifier;
         if (identifier.startsWith("did:factom:")) {
-            chainId = DIDVersion.FACTOM_V1_JSON.getMethodSpecificId(identifier);
+            String methodSpecificId = DIDVersion.FACTOM_V1_JSON.getMethodSpecificId(identifier);
+            String[] parts = methodSpecificId.split(":");
+            if (parts.length > 1 && parts[1].length() == 64) {
```

Comment:
Shouldn't the last split not be the chainId in all cases?

Guess the algo would be to split in : . Start backwards and find the first part with length 64 to make it really future proof

Reply:
Ok, I've updated it to look for strings of the correct length starting from the back. Wasn't sure exactly which error would make sense when the chainId can't be found, so I made a new one.

Reply:
See remark about nullpointer